### PR TITLE
TSDB: snapshot/restore test

### PR DESCRIPTION
**Pull request by nik9000**
Adds a basic test for tsdb with snapshot/restore.

### CODE REVIEW

File: plugins/repository-hdfs/src/test/resources/rest-api-spec/test/hdfs_repository/50_tsdb.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      search:
+        index: test_index
+        body:
+          query:
+            query_string:
+              query: '+@timestamp:"2021-04-28T18:51:04.467Z" +k8s.pod.name:cat'
+
+  - match: {hits.total.value: 1}
+  - match: {hits.hits.0._source.k8s.pod.uid: 947e4ced-1786-4e53-9e0c-5c447e959507}
+  # TODO assert that the _tsid is correct after restore.
```
Comment (nik9000) [inline plugins/repository-hdfs/src/test/resources/rest-api-spec/test/hdfs_repository/50_tsdb.yml:147]: I'd love to use the same trick I use in #78028 to run an agg on `tsid` using a runtime field that looks like `_tsid` just to get the test a little more fleshed out. Just to have *something* we can look at before `_tsid` is generated. But, alas, painless isn't available to these tests.

**Review by imotov — COMMENTED** [review]
I am a bit confused. Why do you only put this into hadoop repo and why hadoop?

**Comment by nik9000** [comment]
I put it into Hadoop so I could run it. I can have another look, but I
didn't see any shared infrastructure for tests against all repos.

**Comment by nik9000** [comment]
> I put it into Hadoop so I could run it. I can have another look, but I didn't see any shared infrastructure for tests against all repos.

I think it'd be more correct for me to run this with the fs repository in rest-api-spec. It'd be cool to have a thing that runs the same tests against all the repos but I don't think that is a "for now" thing.

**Comment by nik9000** [comment]
@imotov could you have another look?

I've moved this into regular rest-api-tests which seems like the simplest thing. It'd be cool to be able to share snapshot/restore tests with a bunch of repos, but I don't think that is a "for now" problem.

**Review by imotov — APPROVED** [review]
LGTM, I think you need to finish the hdfs cleanup though.

**Pull request merged**